### DISCUSSION TITLE
Several modules missing, runtime errors

### DISCUSSION THREAD

**Discussion by MAXimized490** [netbox-community/netbox#15358] [Help Wanted!] [answered]
Hello all! I am following the installation documentation but have run into some issues.

I am currently stuck at testing the application, where I try to run the development server. I first received an error about missing django, which I installed. Then I got several more of these errors, adding them manually as I went on. These included requests, psycopg2, packaging, and netaddr.

After adding all of those, I am now getting the following error which I don't have a clue how to solve. I've obviously gone off the path slightly of the documentation, and I can see that all of these missing packages are included in the requirements.txt, so I'm not sure what went wrong.
```sh
  File "/usr/lib/python3/dist-packages/django/db/models/base.py", line 113, in __new__
    raise RuntimeError(
RuntimeError: Model class django.contrib.contenttypes.models.ContentType doesn't declare an explicit app_label and isn't in an application in INSTALLED_APPS.
```

**Comment by markkuleinio** [comment, marked as answer]
In step https://docs.netbox.dev/en/stable/installation/3-netbox/#create-a-super-user you activated the virtual environment that was created by running upgrade.sh. Now in step https://docs.netbox.dev/en/stable/installation/3-netbox/#test-the-application you still need to be in the virtual environment for the manage.py running to succeed.

(edit: also note the tip "Check that the Python virtual environment is still active before attempting to run the server." in the documentation as well)

--> Activate the venv first

Unfortunately, if you really installed Django and other packages in your system level, your system may be slightly unstable now for the long run, so you may want to consider starting again from a fresh VM. Or, do a reinstall later when you possibly face package mismatches later.

**Reply by candlerb** [reply]
As a way to avoid this problem in future, you can do

```
/opt/netbox/venv/bin/python3 manage.py ...
```

(actually either "python" or "python3" is fine), instead of

```
python3 manage.py ...
```

Running python in the former way will *always* activate the correct venv, so doesn't depend on state from previous commands.

**Reply by MAXimized490** [reply]
That was it! I exited the venv to add the housekeeping cron job and didn't realize that I had to get back into it, nor did I ever need to leave it. Just completed setup and everything is working!
